srml/contract/src/tests.rs

```diff
@@ -111,6 +116,15 @@ impl ContractAddressFor<H256, u64> for DummyContractAddressFor {
 	}
 }
 
+pub struct DummyKeySpaceGenerator;
+impl KeySpaceGenerator<u64> for DummyKeySpaceGenerator {
+	fn key_space(account_id: &u64) -> KeySpace {
```

Comment:
I was looking at the contractid dummy method (a few line above) and thought that it would not need it.
Should we also consider changing `DummyContractAddressFor`?

Reply:
Hm, but `DummyContractAddressFor` and `DummyKeySpaceGenerator` have different requirements, no? As far as I understad, it is ok to implement `ContractAddressFor` and return a non-unique from `DummyContractAddressFor` simply because it doesn't require it. However, `KeySpaceGenerator` has this requirement which `DummyKeySpaceGenerator` doesn't satisfy.

It is not the end of the world, but personally I'd prefer keeping traits to actually satisfy the required contract. In the exceptional cases, if doing so is not possible/feasible, we need to at least leave a comment explaining that why we went this way. Otherwise it might be confusing for the future maintainers.

Reply:
The issue with DummyContractAddressFor is that we can only generate one accountid so it limits test cases.

Reply:
Indeed, this is a downside of `DummyContractAddressFor`.